compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1997,6 +2003,7 @@ public void visit(BLangForeach foreach) {
 
     @Override
     public void visit(BLangFromClause fromClause) {
+        this.queryEnv = fromClause.env;
```

Comment:
Shall we add a comment to explain why we dont have to restore the env to `prevQueryEnv` here

Reply:
Added in b7f8885